jaraco/functools.py

```diff
@@ -304,74 +376,101 @@ def invoke(f, *args, **kwargs):
     Use functools.partial to pass parameters to the initial call
 
     >>> @functools.partial(invoke, name='bingo')
-    ... def func(name): print("called with", name)
+    ... def func(name): print('called with', name)
     called with bingo
     """
     f(*args, **kwargs)
     return f
 
 
-def call_aside(*args, **kwargs):
-    """
-    Deprecated name for invoke.
-    """
-    warnings.warn("call_aside is deprecated, use invoke", DeprecationWarning)
-    return invoke(*args, **kwargs)
+def call_aside(f: Callable[P, R], *args: P.args, **kwargs: P.kwargs) -> Callable[P, R]:
+    """Deprecated name for invoke."""
+    warnings.warn(
+        '`jaraco.functools.call_aside` is deprecated, '
+        'use `jaraco.functools.invoke` instead',
+        DeprecationWarning,
+        stacklevel=2,
+    )
+    return invoke(f, *args, **kwargs)
 
 
-class Throttler:
-    """
-    Rate-limit a function (or other callable)
-    """
+# We can't use P for parameters because of Throttler's descriptor behavior.
+
+
+class Throttler(Generic[R]):
+    """Rate-limit a function (or other callable)."""
 
-    def __init__(self, func, max_rate=float('Inf')):
+    last_called: float
+    func: Callable[..., R]
+
+    def __init__(
+        self,
+        func: Callable[..., R] | Throttler[R],
+        max_rate: SupportsFloat = float('Inf'),
+    ) -> None:
         if isinstance(func, Throttler):
             func = func.func
         self.func = func
-        self.max_rate = max_rate
+        self.max_rate = float(max_rate)
         self.reset()
 
-    def reset(self):
-        self.last_called = 0
+    def reset(self) -> None:
+        self.last_called = 0.0
```

Comment:
Let's avoid changing the implementation when adding type annotations. Let's re-use the return type for `time.time()` and the input type for `time.sleep()`.